src/LSST/state/Model.h

```diff
@@ -143,34 +167,66 @@ class Model {
     // VI-PH updateErrorCodeVI
     // VI-PH writeClosedLoopContolEnablesVI
 
+    // This is not ideal, but the various State children need
+    // access to `Model` members but the mutex `_mtx` has already
+    // been locked. For the sake of consistency, and to prevent
+    // accidental calls to functions where `_mtx` must be locked
+    // before calling, all of the Model members that `State` needs
+    // to call are private.
+    friend class state::State;
+    friend class state::IdleState;
+
 protected:
     // VI-PH startupTDMSLoggingVI
     // VI-PH stopTDMSLoggingVI
     // VI-PH stopCellCommunicationsVI
 
 private:
+
+    /// Go to safe mode, which includes turning off MOTOR and COMM power.
+    /// @param note - Description of reason for safe mode.
+    /// @return true if the system was not already trying to reach.
+    bool _goToSafeMode(std::string const& note);

```

Comment:
Just want to make sure the MOTOR will be turned off first, and followed by COMM.

Reply:
That seems like logic that should be put in the power system. Does MOTOR power need to be all the way on beofe COMM power is on, or is just making sure the MOTOR power on bit is set considered good enough by the current software?  It seems like  wating for the MOTOR power to be on and stable before trying to turn on COMM power would be safer. Adding this logic to the PowerSystem should not be difficult, but should be in another ticket.

Reply:
The M2 machine is designed in the following sequence:

[Power on]
1. Power on the communication power.
2. Once stable, power on the motor power.

[Power off]
1. Power off the motor power.
2. Once done, power off the communication power.

Reply:
I'll make a ticket to put that in the PowerSystem code. I haven't seen this logic in the LabView code.

Reply:
This logic is in the `ts_mtm2` instead of `ts_mtm2_cell`. The `ts_mtm2` has been translated to `ts_m2gui`.

Reply:
After a second thought, maybe it would be easier to test this C++ code on summit without adding this restriction.

Reply:
Duplication of safety items is usually not a bad thing. Making it so that the an error is sent to the GUI if something tries to turn on comm power if motor power is not on or tries to turn off comm power before motor power is off, is fairly simple.